src/Types/Transaction.purs

```diff
@@ -29,6 +35,31 @@ newtype Transaction = Transaction
 
 derive instance newtypeTransaction :: Newtype Transaction _
 
+-- Semigroup for Transaction only appends transactions that are both valid,
+-- if both are false, take the rightmost. I'm not sure if this is the behaviour
+-- we actually want. What determines if something "is_valid" or not?
+instance semigroupTransaction :: Semigroup Transaction where
+  append (Transaction tx) (Transaction tx') =
+    Transaction
+      if tx.is_valid then
+        if tx'.is_valid then
+          { body: tx.body <> tx'.body
+          , witness_set: tx.witness_set <> tx'.witness_set
+          , is_valid: true
```

Comment:
I'm curious what `is_valid` is used for and why it's different in the instance below.

Reply:
Admittedly I'm not sure what determines a "valid" transaction or "invalid" for that matter. I've added a lot of comments in this PR as I'm unsure on a few things, this being one of them. I've made the logic here essentially ignore invalid transactions for the `Monoid` instance - hence why `false` is like the identity. Not sure if this is 100% what we want.